contracts/jobs/libraries/JobLib.sol

```diff
@@ -51,13 +49,23 @@ library JobLib {
             return false;
         }
 
-        if (uint256(keccak256(_claimBlock, block.blockhash(_claimBlock), _segmentNumber)) % _verificationRate == 0) {
+        // Use block hash and block number of the block after a claim to determine if a segment
+        // should be verified
+        if (uint256(keccak256(_claimBlock + 1, block.blockhash(_claimBlock + 1), _segmentNumber)) % _verificationRate == 0) {
```

Comment:
Is the fact that `block.blockhash` is only available for the most recent 256 blocks a problem here? Does it mean there's only a ~64 minute window for verification and slashing to take place before this code wouldn't work?

Reply:
Right...I forgot about that restriction. We probably cannot calculate the block hash in `shouldVerifySegment` and instead have to store the block hash at some point so that it can be used later

Reply:
Ok, need to think about this more...since it seems like this limitation is a problem for us potentially.